diff --git a/node_modules/@headlessui/react/dist/headlessui.esm.js b/node_modules/@headlessui/react/dist/headlessui.esm.js
index 9d9d7b7..d7d6b03 100644
--- a/node_modules/@headlessui/react/dist/headlessui.esm.js
+++ b/node_modules/@headlessui/react/dist/headlessui.esm.js
@@ -1,4 +1,20 @@
-import { useLayoutEffect, useEffect, useState, useRef, useMemo, createElement, createContext, useCallback, useContext, Fragment, isValidElement, cloneElement, forwardRef, useReducer, createRef } from 'react';
+import {
+  cloneElement,
+  createContext,
+  createElement,
+  createRef,
+  forwardRef,
+  Fragment,
+  isValidElement,
+  useCallback,
+  useContext,
+  useEffect,
+  useLayoutEffect,
+  useMemo,
+  useReducer,
+  useRef,
+  useState,
+} from 'react'
 
 function _extends() {
   _extends = Object.assign || function (target) {
@@ -1007,9 +1023,11 @@ var Items = /*#__PURE__*/forwardRefWithAs(function Items(props, ref) {
 
       case Keys.Enter:
         event.preventDefault();
-        dispatch({
-          type: ActionTypes.CloseMenu
-        });
+        if (passthroughProps.closeOnSelect) {
+          dispatch({
+              type: ActionTypes.CloseMenu
+          });
+        }
 
         if (state.activeItemIndex !== null) {
           var _document$getElementB;
@@ -1017,12 +1035,13 @@ var Items = /*#__PURE__*/forwardRefWithAs(function Items(props, ref) {
           var _id = state.items[state.activeItemIndex].id;
           (_document$getElementB = document.getElementById(_id)) === null || _document$getElementB === void 0 ? void 0 : _document$getElementB.click();
         }
+        if(passthroughProps.closeOnSelect) {
+          disposables().nextFrame(function () {
+            var _state$buttonRef$curr2;
 
-        disposables().nextFrame(function () {
-          var _state$buttonRef$curr2;
-
-          return (_state$buttonRef$curr2 = state.buttonRef.current) === null || _state$buttonRef$curr2 === void 0 ? void 0 : _state$buttonRef$curr2.focus();
-        });
+            return (_state$buttonRef$curr2 = state.buttonRef.current) === null || _state$buttonRef$curr2 === void 0 ? void 0 : _state$buttonRef$curr2.focus();
+          });
+        }
         break;
 
       case Keys.ArrowDown:
@@ -1085,7 +1104,7 @@ var Items = /*#__PURE__*/forwardRefWithAs(function Items(props, ref) {
 
         break;
     }
-  }, [dispatch, searchDisposables, state]);
+  }, [dispatch, searchDisposables, state, passthroughProps.closeOnSelect]);
   var propsBag = useMemo(function () {
     return {
       open: state.menuState === MenuStates.Open
@@ -1146,16 +1165,18 @@ function Item(props) {
   }, [bag, id]);
   var handleClick = useCallback(function (event) {
     if (disabled) return event.preventDefault();
-    dispatch({
-      type: ActionTypes.CloseMenu
-    });
-    d.nextFrame(function () {
-      var _state$buttonRef$curr5;
+    if (passthroughProps.closeOnSelect) {
+      dispatch({
+        type: ActionTypes.CloseMenu
+      });
+      d.nextFrame(function () {
+        var _state$buttonRef$curr5;
 
-      return (_state$buttonRef$curr5 = state.buttonRef.current) === null || _state$buttonRef$curr5 === void 0 ? void 0 : _state$buttonRef$curr5.focus();
-    });
+        return (_state$buttonRef$curr5 = state.buttonRef.current) === null || _state$buttonRef$curr5 === void 0 ? void 0 : _state$buttonRef$curr5.focus();
+      });
+    }
     if (onClick) return onClick(event);
-  }, [d, dispatch, state.buttonRef, disabled, onClick]);
+  }, [d, dispatch, state.buttonRef, disabled, onClick, passthroughProps.closeOnSelect]);
   var handleFocus = useCallback(function () {
     if (disabled) return dispatch({
       type: ActionTypes.GoToItem,
